plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -258,13 +262,17 @@ func (m *MQTTConsumer) recvMessage(c mqtt.Client, msg mqtt.Message) {
 		select {
 		case track := <-m.acc.Delivered():
 			<-m.sem
+			m.messagesMutex.RLock()
 			_, ok := m.messages[track.ID()]
+			m.messagesMutex.RUnlock()
 			if !ok {
 				// Added by a previous connection
 				continue
 			}
 			// No ack, MQTT does not support durable handling
+			m.messagesMutex.Lock()
 			delete(m.messages, track.ID())
+			m.messagesMutex.Unlock()
```

Comment:
Doing
```suggestion
			m.messagesMutex.RLock()
			_, ok := m.messages[track.ID()]
			if ok {
				// No ack, MQTT does not support durable handling
				delete(m.messages, track.ID())
			}
			m.messagesMutex.Unlock()
```
might also work but is holding the mutex a bit longer... On the other hand it prevents that exactly this ID is deleted in the meantime...